osf/metrics/reports.py

```diff
@@ -28,13 +30,51 @@ class Meta:
         source = metrics.MetaField(enabled=True)
 
 
-@receiver(pre_save)
+class YearmonthField(metrics.Date):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs, format='strict_year_month', required=True)
+
+    def deserialize(self, data):
+        if isinstance(data, YearMonth):
+            return data
+        elif isinstance(data, str):
+            return YearMonth.from_str(data)
+        elif isinstance(data, (datetime.datetime, datetime.date)):
+            return YearMonth.from_date(data)
+        else:
+            raise ValueError('unsure how to deserialize "{data}" (of type {type(data)}) to YearMonth')
+
+    def serialize(self, data):
+        if isinstance(data, str):
+            return data
+        elif isinstance(data, YearMonth):
+            return str(data)
+        elif isinstance(data, (datetime.datetime, datetime.date)):
```

Comment:
nit: datetime is a subclass of date, so can just be `isinstance(data, datetime.date)`